packages/api/db/migration/20240524031314_create_animal_identifier_table_and_modify_identifier_placement_table.js

```diff
@@ -0,0 +1,125 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+
+export const up = async function (knex) {
+  await knex.schema.createTable('animal_identifier_type', (table) => {
+    table.increments('id').primary();
+    table.string('key').notNullable();
+  });
+
+  await knex.raw(`
+    CREATE TEMPORARY TABLE temp AS
+    SELECT * FROM animal_identifier_placement;
+  `);
+
+  await knex.schema.alterTable('animal', (t) => {
+    t.dropForeign('identifier_placement_id');
+  });
+
+  await knex.schema.dropTable('animal_identifier_placement');
+
+  await knex.schema.createTable('animal_identifier_placement', (table) => {
+    table.increments('id').primary();
```

Comment:
For Nav, to me this looks exactly as described in the Jira ticket at least!

@SayakaOno I am not sure if I remember discussing this. Are you suggesting a relationship table(between type and placement)? Or two ids on the animal (one for type one for placement)?

My understanding is how this was meant to work is that a user will react select an indentifier_type (Ear_tag) then we will filter the identifier_placements (left ear, right ear) based on that original choice for a second react select. The placement_id is added I think because combination (ear_tag) and (left_leg) don't make sense.

The architecture we have chosen seems not great I agree. I would prefer some composite key enum like: 

![Screenshot 2024-08-02 at 10 44 24 AM](https://github.com/user-attachments/assets/bfa9b4fe-114c-4b10-ac90-65717baf1edb)

Or just omit the "left ear" , "right_ear" selection altogether -- "ear tag" or "leg_band" seems plenty descriptive.

Reply:
Thank you for your input Duncan!

Nav, we discussed this with the team, and the decision was made to remove the tag placement input.
![Screenshot 2024-08-06 at 10 52 04 AM](https://github.com/user-attachments/assets/5f36d4f9-7c1f-4c7b-86b4-191b41b769b3)

This means that we won't need the `animal_identifier_placement` table. I'm terribly sorry for the update on the requirement, and for the lack of clarity in the initial requirements as well.

In addition to dropping the `animal_identifier_placement` table, the expected changes are:
- remove `identifier_placement_id` column from `animal` table
- update models
- update tests
- update routes
- other references?

Since this requires a substantial amount of changes, we could create another ticket/PR for this. (In this case, we could have one PR for `animal_identifier_type`, and another for removing `animal_identifier_placement`.)
Please let us know what works best.

Reply:
@SayakaOno That sounds reasonable and I think separate PR's for each requirement would be better.